### PR TITLE
Fix torchscript tests for GPT-NeoX

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Fix torchscript tests for GPT-NeoX. The main issue comes from the fact that current `RotaryEmbedding` changes the model structure in `forward`.

This PR creates the necessary embeddings in `__init__`, which basically makes the cache (of embedding) mechanism useless. Furthermore, the attribute names seems a bit confusing now. We could probably add some attribute (ex. `init_sin_cos_cache_seq_len`) in config with a value `<= max_position_embeddings`, but I think it's way too much.

Not certain if it is worth it. However, with a PR opened, we have a reference.

The current failing test is 
https://github.com/huggingface/transformers/runs/7216768053?check_suite_focus=true

### CODE REVIEW

File: src/transformers/models/gpt_neox/modeling_gpt_neox.py [hunk 1]
```diff
@@ -207,7 +209,8 @@ def _attn(self, query, key, value, attention_mask=None, head_mask=None):
             query,
             key.transpose(1, 2),
             beta=1.0,
-            alpha=(1.0 / self.norm_factor),
+            alpha=(torch.tensor(1.0, dtype=self.norm_factor.dtype, device=self.norm_factor.device) / self.norm_factor),
+            # alpha=(1.0 / self.norm_factor),
```
Comment (sgugger) [inline src/transformers/models/gpt_neox/modeling_gpt_neox.py:213]: Should be cleaned up.

**Review by sgugger — APPROVED** [review]
LGTM, thanks for fixing!

**Review by patrickvonplaten — APPROVED** [review]
LGTM!

However, could we add the failing test for reference or do we need to add a new test here?

**Comment by ydshieh** [comment]
> LGTM!
> 
> However, could we add the failing test for reference or do we need to add a new test here?

I updated the PR description to include the current failing test. Regarding new tests, I don't think it's necessary, as we just build the necessary tensors in `__init__` instead of in `forward`, and the current set of tests should be enough :-)

(however, let me know if you have some idea of new necessary test cases!)

**Review by LysandreJik — APPROVED** [review]
LGTM

**Pull request merged**